app/src/main/java/org/gnucash/android/importer/GncXmlHandler.java

```diff
@@ -688,14 +688,14 @@ public void endElement(String uri, String localName, String qualifiedName) throw
                 break;
             case GncXmlHelper.TAG_TRANSACTION:
                 mTransaction.setTemplate(mInTemplates);
-                Split imbSplit = mTransaction.createAutoBalanceSplit();
-                if (imbSplit != null) {
-                    mAutoBalanceSplits.add(imbSplit);
-                }
                 if (mInTemplates){
                     if (!mIgnoreTemplateTransaction)
                         mTemplateTransactions.add(mTransaction);
                 } else {
+                    Split imbSplit = mTransaction.createAutoBalanceSplit();
+                    if (imbSplit != null) {
+                        mAutoBalanceSplits.add(imbSplit);
+                    }
```

Comment:
If we get a transaction with unbalanced splits (scheduled or normal), I'd say it's either we've failed to import it correctly or there's something wrong in the file.